cpp/oneapi/dal/algo/rbf_kernel/backend/gpu/compute_kernel_dense_dpc.cpp

```diff
@@ -29,46 +27,112 @@ using input_t = compute_input<task::compute>;
 using result_t = compute_result<task::compute>;
 using descriptor_t = detail::descriptor_base<task::compute>;
 
-namespace daal_rbf_kernel = daal::algorithms::kernel_function::rbf;
-namespace interop = dal::backend::interop;
+namespace pr = dal::backend::primitives;
+
+template <typename Float>
+inline auto compute_rbf_values(sycl::queue& queue,
+                               const pr::ndview<Float, 1>& sqr_x_nd,
+                               const pr::ndview<Float, 1>& sqr_y_nd,
+                               pr::ndview<Float, 2>& res_nd,
+                               double sigma,
+                               const dal::backend::event_vector& deps = {}) {
+    const std::int64_t x_row_count = sqr_x_nd.get_dimension(0);
+    const std::int64_t y_row_count = sqr_y_nd.get_dimension(0);
+    ONEDAL_ASSERT(res_nd.get_count() == x_row_count * y_row_count);
+
+    const Float coeff = static_cast<Float>(-0.5 / (sigma * sigma));
+
+    const Float* sqr_x_ptr = sqr_x_nd.get_data();
+    const Float* sqr_y_ptr = sqr_y_nd.get_data();
+    Float* res_ptr = res_nd.get_mutable_data();
+
+    const Float threshold = dal::backend::exp_low_threshold<Float>();
+
+    const auto wg_size = dal::backend::propose_wg_size(queue);
+    const auto range =
+        dal::backend::make_multiple_nd_range_2d({ x_row_count, y_row_count }, { wg_size, 1 });
+
+    const std::size_t ld = y_row_count;
```

Comment:
Suggest to move these declaration to the `submut` body to not pollute scope of the function